### DISCUSSION TITLE
React Native offline-first

### DISCUSSION THREAD

**Discussion by iromjkee** [urql-graphql/urql#1549] [Q&A] [answered]
Hi guys, could you please help? Just wanna know if it`s possible to use AsyncStorage of React Native with urlq for storing data while app offline? If yes, may be you have a few examples of it? Started developing my app with Apollo, but researched a bit and found **urlq**, that seems to be great for my app and much better than apollo is.

**Comment by JoviDeCroock** [comment, marked as answer]
Hey @iromjkee 

In the [normalized cache docs](https://formidable.com/open-source/urql/docs/graphcache/offline/#setup) we talk about the offline-exchange and that it allows for a custom-storage.

The default we offer is an idb one but we can use that one as an example to expand into `AsyncStorage` 

- [replace with network detection for RN](https://github.com/FormidableLabs/urql/blob/main/exchanges/graphcache/src/default-storage/index.ts#L197)
- [need to become their AsyncStorage equivalents](https://github.com/FormidableLabs/urql/blob/main/exchanges/graphcache/src/default-storage/index.ts#L98-L195)
